inc/class-wpseo-statistics.php

```diff
@@ -127,6 +125,11 @@ private function get_post_count( $seo_ranking ) {
 
 		$posts['fields']      = 'ids';
 		$posts['post_status'] = 'publish';
+
+		if ( current_user_can( 'moderate_comments' ) === false ) {
+			$posts['author'] = get_current_user_id();
```

Comment:
If you introduce a method for getting the user id, be sure you use it consequently.